### ISSUE TITLE
Two tests in src/test/perl/partition_gpt.t fail on GitHub runners

### ISSUE THREAD

**Issue by jrha** [quattor/ncm-lib-blockdevices#100]
The failing tests are checking the exit code of create and begin:
```perl
my $mpath1 = NCM::Partition->new ("/system/blockdevices/partitions/".escape("mapper/abcdef123p1"), $cfg);
is ($mpath1->create, 0, "Partition $mpath1->{devname} on logical partitions test created correctly");
is ($mpath1->begin, 0, 'Begin from 0 (no offset) for 1st mpath partition');
diag explain $mpath1;
```

```
not ok 26 - Perl warning: Use of uninitialized value in join or string at /__w/ncm-lib-blockdevices/ncm-lib-blockdevices/target/dependency/build-scripts/Test/Quattor.pm line 348.
# 
not ok 27 - Perl warning: Use of uninitialized value $p in concatenation (.) or string at /__w/ncm-lib-blockdevices/ncm-lib-blockdevices/target/lib/perl/NCM/Blockdevices.pm line 511.
Perl warning: Use of uninitialized value in join or string at /__w/ncm-lib-blockdevices/ncm-lib-blockdevices/target/dependency/build-scripts/Test/Quattor.pm line 348.
```

Both of these seem to be `uninitialized value` warnings becoming hard errors.

One is from the following in `NCM::Blockdevices`:
```perl
    my $p = abs_path($self->devpath);
    my $f =  CAF::Process->new([FILES, $p], log => $self)->output();

    $self->debug(4, "Checking for filesystem on device $p",
                        " with regexp '$fsregex' in output $f.");
```

The other two both appear to be caused by the same line in the mocked `output`/`toutput` methods in `Test::Quattor`:
```perl
        my $cmd = join(" ", @{$self->{COMMAND}});
```

Perl is being installed from exactly the same RPM as on our SL7 nightlies (currently `perl-5.16.3-299.el7_9.x86_64`)  which do not exhibit this issue.

To me this looks like `$self` is ending up undefined? Perhaps some unforeseen environment difference?

**Comment by wpoely86** [comment]
The github tests are run inside a centos7 container which is recreated on every run so the environment is fully reproducible. Last time I tried I could also not get these tests to run on my Fedora. So I wonder if there is anything special about your sl7 box?

In any case we need to migrate to some el8 flavour before the end of the year.

**Comment by jrha** [comment]
This appears to have been fixed by switching to an EL8 based image in #107, there is another test which appears to have a rare race-condition, but I'll open a new issue for that.
